gradle/instrumentation.gradle

```diff
@@ -40,6 +40,12 @@ afterEvaluate {
       // OpenTelemetry SDK is not needed for compilation, and :opentelemetry-sdk-shaded-for-testing
       // is brought in for tests by project(:testing-common) below
       exclude group: 'io.opentelemetry', module: 'opentelemetry-sdk'
+      // need explicit excludes on transitive dependencies because they are included explicitly
```

Comment:
Nice find - I think we should use dependency constraints instead of explicitly depending to make sure we don't forget adding this anywhere

https://docs.gradle.org/current/userguide/dependency_constraints.html#sec:adding-constraints-transitive-deps